qiskit_dynamics/models/lindblad_model.py

```diff
@@ -0,0 +1,616 @@
+        validate: bool = True,
+    ):
+        """Initialize.
+
+        Args:
+            static_hamiltonian: Constant term in Hamiltonian.
+            hamiltonian_operators: List of operators in Hamiltonian with time-dependent
+                                   coefficients.
+            hamiltonian_signals: Time-dependent coefficients for hamiltonian_operators.
+            static_dissipators: List of dissipators with coefficient 1.
+            dissipator_operators: List of dissipator operators with time-dependent coefficients.
+            dissipator_signals: Time-dependent coefficients for dissipator_operators.
+            rotating_frame: Rotating frame in which calcualtions are to be done.
+                            If provided, it is assumed that all operators were
+                            already in the frame basis.
+            evaluation_mode: Evaluation mode to use. See ``LindbladModel.evaluation_mode``
+                             for more details.
+            validate: If True check input hamiltonian_operators and static_hamiltonian are
+                      Hermitian.
+
+        Raises:
+            QiskitError: If model insufficiently or incorrectly specified.
+        """
+
+        if (
+            static_hamiltonian is None
+            and hamiltonian_operators is None
+            and static_dissipators is None
+            and dissipator_operators is None
+        ):
+            raise QiskitError(
+                self.__class__.__name__
+                + """ requires at least one of static_hamiltonian,
+                              hamiltonian_operators, static_dissipators, or dissipator_operators
+                              to be specified at construction."""
+            )
+
+        static_hamiltonian = to_numeric_matrix_type(static_hamiltonian)
+        hamiltonian_operators = to_numeric_matrix_type(hamiltonian_operators)
+        if validate:
+            if (hamiltonian_operators is not None) and (not is_hermitian(hamiltonian_operators)):
+                raise QiskitError("""LinbladModel hamiltonian_operators must be Hermitian.""")
+            if (static_hamiltonian is not None) and (not is_hermitian(static_hamiltonian)):
+                raise QiskitError("""LinbladModel static_hamiltonian must be Hermitian.""")
+
+        self._operator_collection = self.construct_operator_collection(
+            evaluation_mode=evaluation_mode,
+            static_hamiltonian=static_hamiltonian,
+            hamiltonian_operators=hamiltonian_operators,
+            static_dissipators=static_dissipators,
+            dissipator_operators=dissipator_operators,
+        )
+        self._evaluation_mode = evaluation_mode
+        self.vectorized_operators = "vectorized" in evaluation_mode
+
+        self._rotating_frame = None
+        self.rotating_frame = rotating_frame
+
+        self.signals = (hamiltonian_signals, dissipator_signals)
+
+    @classmethod
+    def from_hamiltonian(
```

Comment:
You could also add an optional `hamiltonian` kwarg to init to implement this as part of regular init? Maybe getting a lot of args for init at this point though...

Reply:
I think I'd like to keep this separate. If we add it to the init we'll need to add validation to check if redundant/inconsistent information is being passed and document the multiple modes of behaviour of the init. Not that huge a deal but it feels simpler to keep it structurally separate.